pkg/virtualgateway/enqueue_requests_for_mesh_events.go

```diff
@@ -0,0 +1,71 @@
+package virtualgateway
+
+import (
+	"context"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/k8s"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/mesh"
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+func NewEnqueueRequestsForMeshEvents(k8sClient client.Client, log logr.Logger) *enqueueRequestsForMeshEvents {
+	return &enqueueRequestsForMeshEvents{
+		k8sClient: k8sClient,
+		log:       log,
+	}
+}
+
+var _ handler.EventHandler = (*enqueueRequestsForMeshEvents)(nil)
+
+type enqueueRequestsForMeshEvents struct {
+	k8sClient client.Client
+	log       logr.Logger
+}
+
+// Create is called in response to an create event
+func (h *enqueueRequestsForMeshEvents) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
+	// no-op
+}
+
+// Update is called in response to an update event
+func (h *enqueueRequestsForMeshEvents) Update(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
+	// virtualGateway reconcile depends on mesh is active or not.
+	// so we only need to trigger virtualGateway reconcile if mesh's active status changed.
+	msOld := e.ObjectOld.(*appmesh.Mesh)
+	msNew := e.ObjectNew.(*appmesh.Mesh)
+
+	if mesh.IsMeshActive(msOld) != mesh.IsMeshActive(msNew) {
```

Comment:
Should we also check for a change in namespace selector for enqueuing the corresponding virtualgateway?

Reply:
I think it should be fine as the virtual gateway still points to the same mesh

Although side note: it brings up an interesting point what happens when we try to point a mesh with active resources (virtualnodes etc) to a new namespace on the fly. Worth investigating to understand the behavior and manage accordingly. I've taken a note